src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureUnitTests.cs

```diff
@@ -382,8 +372,7 @@ public async Task TestAbstractClass()
             var csInput = @" 
         using System.Runtime.Versioning; using System;
         namespace Preview_Feature_Scratch
-        {" +
-@"
+        {
 
             class {|#0:Program|} : AbClass
```

Comment:
This is the approach I tried first, but I wasn't sure how we can use the IOperation based APIs to mark `AbClass` here. This is what I have currently:
``` csharp
            INamedTypeSymbol? baseType = symbol.BaseType; // This will return AbClass. symbol is still Program
            if (baseType != null)
            {
                if (SymbolIsAnnotatedAsPreview(baseType, requiresPreviewFeaturesSymbols, previewFeatureAttributeSymbol))
                {
                    context.ReportDiagnostic(symbol.CreateDiagnostic(DerivesFromPreviewClassRule, baseType.Name));
                }
            }
```

So the diagnostic will say `'Program' derives from preview class 'AbClass' and therefore needs to opt into preview features. See https://aka.ms/dotnet-warnings/preview-features for more information.`

Reply:
I wonder if @Youssef1313 knows how we can access the `AbClass` symbol here. As mentioned on another comment, it's important that we find a way to get the diagnostic to paint on the base class / interface symbol instead of the type symbol as that is needed to give `#pragma warning disable` around the preview symbol instead of the consuming symbol.

Reply:
What is the problem with having the diagnostic on the base class though? Suppressing the diagnostic on the base class is no different than suppressing it on the interface list right?

Reply:
It's a difference between:

**Desired:**

```csharp
    public readonly struct Int32 : IComparable, IConvertible, ISpanFormattable, IComparable<int>, IEquatable<int>
#if FEATURE_GENERIC_MATH
#pragma warning disable SA1001, CA2252 // comma formatting and preview features
        , IBinaryInteger<int>,
          IMinMaxValue<int>,
          ISignedNumber<int>
#pragma warning restore SA1001, CA2252
#endif // FEATURE_GENERIC_MATH
    {
```

And **Current:**

```csharp
#pragma warning disable CA2252 // preview features
    public readonly struct Int32 : IComparable, IConvertible, ISpanFormattable, IComparable<int>, IEquatable<int>
#pragma warning restore CA2252
#if FEATURE_GENERIC_MATH
#pragma warning disable SA1001 // comma formatting
        , IBinaryInteger<int>,
          IMinMaxValue<int>,
          ISignedNumber<int>
#pragma warning restore SA1001
#endif // FEATURE_GENERIC_MATH
    {
```

With the current behavior, we don't have a granularity safeguard around _which_ interfaces we're opting into using as preview.

Reply:
Separately, @sharwell, @Youssef1313 , @jmarolf, using the IOperation based APIs, is there a way to mark `AbClass` with a diagnostic in the following example:
``` csharp
class Program : [|AbClass|] //AbClass is an abstract class defined elsewhere
{

}
```

Reply:
Since each language expresses the concept of inheritance differently I believe you will need to get the [Syntax](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.ioperation.syntax?view=roslyn-dotnet-3.10.0#Microsoft_CodeAnalysis_IOperation_Syntax) from the operation node and then walk the syntax tree for each language to get this location.

Reply:
@jmarolf : Just to double check, this means I need to abandon the current implementation and start over for a C# one right? AFAIK, there's no way to do 95% of the work in IOperation and drop down to C#/VB for the last 5%?

Reply:
You can make your current class abstract with single abstract helper method `FindInheritanceLocation` that a C# and VB impl can implement

Reply:
cc @333fred for potential enhancements to IOperation APIs.